File: 백준/Silver/10866. 덱/덱.py
from collections import deque
import sys

data = deque()
n = int(sys.stdin.readline().rstrip())
for _ in range(n):
  s = sys.stdin.readline().rstrip().split()
  
  if s[0] == "push_front":
    data.appendleft(int(s[1]))
  elif s[0] == "push_back":
    data.append(int(s[1]))
  elif s[0] == "pop_front":
    if len(data) != 0:
      print(data.popleft())
    else:
      print(-1)
  elif s[0] == "pop_back":
    if len(data) != 0:
      print(data.pop())
    else:
      print(-1)
  elif s[0] == "size":
    print(len(data))
  elif s[0] == "empty":
    if len(data) == 0:
      print(1)
    else:
      print(0)
  elif s[0] == "front":
    if len(data) != 0:
      print(data[0])
    else:
      print(-1)
  elif s[0] == "back":
    if len(data) != 0:
      print(data[len(data)-1])
    else:
      print(-1)

File: 백준/Silver/10866. 덱/덱.java
import java.io.*;
import java.util.LinkedList;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));
        //삭제, 추가가 빈번하므로 likedList 사용
        LinkedList<Integer> lst = new LinkedList<Integer>();
        int num = Integer.parseInt(bf.readLine());
        for (int i = 0; i < num; i++) {
            String[] inputs = bf.readLine().split(" ");
            String command = inputs[0];
            Integer n = null;
            if (inputs.length == 2) {
                n = Integer.valueOf(inputs[1]);
            }
            if ("push_back".equals(command)) {
                lst.add(n);
            } else if("push_front".equals(command)){
                lst.addFirst(n);
            } else if (("front").equals(command)) {
                Integer number = lst.peekFirst() != null ? lst.peekFirst() : -1;
                System.out.println(number);
            } else if (("back").equals(command)) {
                Integer number = lst.peekLast() != null ? lst.peekLast() : -1;
                System.out.println(number);
            } else if (("size").equals(command)) {
                System.out.println(lst.size());
            } else if (("empty").equals(command)) {
                Integer number = lst.isEmpty() ? 1 : 0;
                System.out.println(number);
            } else if("pop_front".equals(command)){
                Integer number = lst.pollFirst();
                if (null == number) {
                    System.out.println(-1);
                } else {
                    System.out.println(number);
                }
            } else if("pop_back".equals(command)){
                Integer number = lst.pollLast();
                if (null == number) {
                    System.out.println(-1);
                } else {
                    System.out.println(number);
                }
            }
        }

    }

}

// Scanner sc = new Scanner(new InputStreamReader(System.in));
// BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));
